recipes/default.rb

```diff
@@ -23,25 +24,52 @@
   group node['awslogs_agent']['group']
 end
 
-
 python_package "awscli-cwlogs" do
   virtualenv node['awslogs_agent']['path']
   version node['awslogs_agent']['version']
   options "--extra-index-url=#{node['awslogs_agent']['plugin_url']}"
 end
 
-template "/etc/init/#{node['awslogs_agent']['service']}.conf" do
+using_systemd = false
+destination_folder = ''
+source_file = ''
+if  'ubuntu' == node['platform']
+  if Chef::VersionConstraint.new('>= 15.04').include?(node['platform_version'])
+    using_systemd = true
+    destination_folder = "/lib/systemd/system/#{node['awslogs_agent']['service']}.service"
+    source_file = "etc/systemd/awslogs_daemon.service.erb"
+  elsif Chef::VersionConstraint.new('>= 12.04').include?(node['platform_version'])
```

Comment:
What if the platform is less than 12.04? I suppose they have more problems than this but still... you should always be covered for all known scenarios.